onnxruntime/core/framework/allocator.cc

```diff
@@ -51,7 +51,12 @@ std::ostream& operator<<(std::ostream& out, const OrtAllocatorInfo& info) {
 
 ORT_API_STATUS_IMPL(OrtCreateAllocatorInfo, _In_ const char* name1, OrtAllocatorType type, int id1,
                     OrtMemType mem_type1, _Out_ OrtAllocatorInfo** out) {
-  *out = new OrtAllocatorInfo(name1, type, id1, mem_type1);
+  if (strcmp(name1, onnxruntime::CPU) == 0) {
+    *out = new OrtAllocatorInfo(name1, type, OrtDevice(), id1, mem_type1);
+  }
+  // Other cases not handled as,
+  // 1. there should be no one calling this function to create non-cpu allocation information.
+  // 2. this function should/will be deprecated.
```

Comment:
This function is needed for creating a tensor with preallocated buffer. In most of the cases, the tensor is on CPU. But, it still could be on GPU.

Reply:
IIRC the WinML folks require the ability to call Run with GPU based inputs. Do they need this API to do that?

---
In reply to: [296057994](https://github.com/microsoft/onnxruntime/pull/1268#discussion_r296057994) [](ancestors = 296057994)

Reply:
I think this is for user calling ORT, the inputs should be always in CPU memory firstly?

Reply:
added the other two cases.